scripts/src/java/org/oppia/android/scripts/lint/LintAnalysisReporter.kt

```diff
@@ -138,6 +164,178 @@ class LintAnalysisReporter {
+
+      val groupedByIssueId = issuesInSeverity.groupBy { it.id }.toSortedMap()
+
+      groupedByIssueId.forEach { (issueId, issuesForId) ->
+        val representativeIssue = issuesForId.first()
+
+        println("\n$BOLD Issue ID: $issueId$RESET")
+        println("  ${colorizeSeverity(representativeIssue.severity)}")
+
+        val allLocations = issuesForId.flatMap { it.locations }
+          .distinctBy { "${it.file}:${it.lineNumber}" }
+          .sortedWith(compareBy({ it.file }, { it.lineNumber.toIntOrNull() ?: 0 }))
+
+        if (allLocations.size == 1) {
+          val location = allLocations.first()
+          println("  File: ${location.file}")
+          if (location.lineNumber.isNotBlank()) {
+            println("  Line: ${location.lineNumber}")
+          }
+        } else {
+          println("  Locations:")
+          allLocations.forEachIndexed { index, location ->
+            println("    ${index + 1}. File: ${location.file}")
+            if (location.lineNumber.isNotBlank()) {
+              println("       Line: ${location.lineNumber}")
+            }
+          }
+        }
+        printIssueBasicInfo(representativeIssue)
+
+        println("-".repeat(58))
+      }
+    }
+  }
+
+  /** Prints issues grouped by file path. */
+  private fun printGroupedByFilePath(issues: List<LintIssue>) {
+    val fileToIssueLocationMap = mutableMapOf<String, MutableList<Pair<LintIssue, LintLocation>>>()
+
+    issues.forEach { issue ->
+      issue.locations.forEach { location ->
+        fileToIssueLocationMap.getOrPut(location.file) { mutableListOf() }
+          .add(Pair(issue, location))
+      }
+    }
+
+    val sortedFiles = fileToIssueLocationMap.keys.sorted()
+
+    sortedFiles.forEach { filePath ->
+      val issueLocationPairs = fileToIssueLocationMap[filePath] ?: emptyList()
+
+      println("\n${"=".repeat(80)}")
+      println("${BOLD}FILE: $filePath (${issueLocationPairs.size} issues)$RESET")
+      println("=".repeat(80))
+
+      val sortedByLine = issueLocationPairs.sortedWith(
+        compareBy<Pair<LintIssue, LintLocation>> { it.second.lineNumber.toIntOrNull() ?: 0 }
+          .thenBy { it.first.severity.ordinal }
+          .thenBy { it.first.id }
+      )
+
+      sortedByLine.forEachIndexed { index, (issue, location) ->
+        println("\n$BOLD Issue #${index + 1}: ${issue.id}$RESET")
+        println("  ${colorizeSeverity(issue.severity)}")
+        if (location.lineNumber.isNotBlank()) {
+          println("  Line: ${location.lineNumber}")
+        }
+        printIssueBasicInfo(issue, indent = "  ")
+        println("-".repeat(58))
+      }
+    }
+  }
+
+  /** Prints basic information about an issue. */
+  private fun printIssueBasicInfo(issue: LintIssue, indent: String = "  ") {
+
+    if (issue.errorLine1.isNotBlank()) {
+      println("${indent}Error Line: ${issue.errorLine1}")
+      if (issue.errorLine2.isNotBlank()) {
+        println("$indent            ${issue.errorLine2}")
```

Comment:
Can we replace the 12-space indentation with a named constant? Something like: private val ERROR_LINE_INDENT = " ".repeat("Error Line: ".length)`